app/javascript/components/star-rating.vue

```diff
@@ -0,0 +1,46 @@
+
+<script setup lang="ts">
+
+type Props = {
+  rating: number | string
+}
+
+const MAX_STARS = 5;
+
+const props = defineProps<Props>();
+
+function halfStars() {
+  return props.rating % 1 !== 0;
+}
+
+function fullStars() {
+  return parseInt(props.rating, 10);
```

Comment:
🚫 **[tsc]** <2345> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Argument of type 'string | number' is not assignable to parameter of type 'string'.
Type 'number' is not assignable to type 'string'.

Reply:
al final se pasara a float el tipo de dato. Equivalente a number en TS